core/adapters/quotient_test.go

```diff
@@ -0,0 +1,102 @@
+package adapters_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"chainlink/core/adapters"
+	"chainlink/core/internal/cltest"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQuotient_Perform_Success(t *testing.T) {
+	tests := []struct {
+		name   string
+		params string
```

Comment:
What about having the 'params' type directly defined as Quotient?  It would be clearer about its purpose and make its syntax accessible at compile time

Reply:
If you look at line 45 we test unmarshalling it into the adapters.Quotient struct.

If we provide it with the unmarshalled version directly we lose this test coverage.